pkg/controller/controller.go

```diff
@@ -631,7 +653,17 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		},
 	}
 
-	if claim.Spec.DataSource != nil && (rc.clone || rc.snapshot) {
+	if claim.Spec.DataSourceRef != nil && claim.Spec.DataSourceRef.Namespace != nil && len(*claim.Spec.DataSourceRef.Namespace) > 0 {
+		if utilfeature.DefaultFeatureGate.Enabled(features.CrossNamespaceVolumeDataSource) && rc.snapshot {
+			volumeContentSource, err := p.getVolumeContentSourceFromXnsDataSource(ctx, claim, sc)
```

Comment:
Do we need a new function, or can we modify `getVolumeContentSource`?

Reply:
Yes. Is possible. Fixed code https://github.com/kubernetes-csi/external-provisioner/pull/805/commits/1f9df93e646bbd126bf7bcc172a7e2a3a2da4e8e
Could you check code?